### PR TITLE
Add Children's Book Test (CBT) dataset

### PR DESCRIPTION
**Pull request by thomwolf**
Add the Children's Book Test (CBT) from Facebook (Hill et al. 2016).

Sentence completion given a few sentences as context from a children's book.

### CODE REVIEW

File: datasets/cbt/cbt.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+        pattern = re.compile(r"(\d+)\s([^\t]+)\t?([^\t]*)?\t*([^\t]*)?")
+
+        if not isinstance(filepath, (list, tuple)):
+            filepath = [filepath]
+
+        for file in filepath:
+            with open(file) as f:
```
Comment (lhoestq) [inline datasets/cbt/cbt.py:195]:
```suggestion
            with open(file, encoding="utf-8") as f:
```

**Review by lhoestq — COMMENTED** [review]
The dummy data can probably be lighter if we remove some unnecessary data files ?
For example by removing the files of `all_questions` in the `books` dummy_data.zip ? and doing so for the other zip files as well

File: datasets/cbt/cbt.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="https://research.fb.com/downloads/babi/",
```
Comment (lhoestq) [inline datasets/cbt/cbt.py:130]: wrong homepage here

**Comment by gchhablani** [comment]
Hi @lhoestq,

I guess this PR can be closed since we merged #2044?

I have used the same link for the homepage, as it is where the dataset is provided, hope that is okay?

**Comment by lhoestq** [comment]
Closing in favor of #2044, thanks again :)

> I have used the same link for the homepage, as it is where the dataset is provided, hope that is okay?

Yea it's ok actually, at that time I thought there was another homepage for this dataset

**Pull request closed without merging**